src/de.hpi.swa.graal.squeak.test/src/de/hpi/swa/graal/squeak/test/AbstractSqueakTestCaseWithImage.java

```diff
@@ -77,6 +78,22 @@ private static void patchImageForTesting() {
         }
     }
 
+    private static void createProcessForTesting() {
+        final PointersObject newProcess = new PointersObject(image, image.processClass);
+        newProcess.instVarAtPut0Slow(PROCESS.PRIORITY, USER_PRIORITY_LIST_INDEX + 1);
+
+        final ArrayObject lists = (ArrayObject) image.getScheduler().instVarAt0Slow(PROCESS_SCHEDULER.PROCESS_LISTS);
+        final PointersObject priority40List = (PointersObject) lists.getObjectStorage()[USER_PRIORITY_LIST_INDEX];
+        final Object firstLink = priority40List.instVarAt0Slow(LINKED_LIST.FIRST_LINK);
+        priority40List.instVarAtPut0Slow(LINKED_LIST.FIRST_LINK, newProcess);
+        if (firstLink == NilObject.SINGLETON) {
+            priority40List.instVarAtPut0Slow(LINKED_LIST.LAST_LINK, newProcess);
+        } else {
+            newProcess.instVarAtPut0Slow(PROCESS.NEXT_LINK, firstLink);
+        }
+        newProcess.instVarAtPut0Slow(PROCESS.LIST, priority40List);
+    }
```

Comment:
Why do you create a new process here for testing? More importantly, the process created here is incomplete (it doesn't have a `suspendedContext` for example), so you'll get an error in case this process is ever selected by the scheduler.

Reply:
I saw some errors in the tests (I think it must have been DebuggerTests>>test04.. and test05... where new processes are created and set to priority Processor activeProcess priority - 1. They were failing because the priority was 9, below the lowestPriority. The only process at 10 is the idleLoop, so I assumed that the active process for the doItContext was the idleLoop. Which could happen if the image was not reloaded. But I did miss setting my new process to be the active process. Anyway, this should be done only if the active process at the beginning of the evaluation is the idleLoop

Reply:
Regarding your general comment, I agree, I got carried away - initially I just wanted to get the initial PR to pass the tests, but as I started to look at all those tests and discovering new things, I kept adding to it.